### PR TITLE
Update documentation and change name to Hardware Wallet Interface

### PR DESCRIPTION
**Pull request by achow101**
Updates the docs to reflect current behavior and describe the stuff that is currently in the repo.

Also renames the project from Hardware Wallet Interaction scripts to Hardware Wallet Interface.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,25 @@ Python software can use the provided library (`hwilib`). Software in other langu
...
+
+For macOS:
+```
+brew install libusb
+```
+
+This project uses the [Poetry](https://github.com/sdispater/poetry) dependency manager. HWI and its dependencies can be installed via poetry with:
+
+```
+poetry install
```
Comment (instagibbs) [inline README.md]: `poetry install hwi` ?

Comment (achow101) [reply README.md]: No, you run that from within the source tree itself. I've clarified this.

File: docs/bitcoin-core-usage.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ $ ../bitcoin/src/bitcoin-cli createwallet "coldcard" true
   "name": "coldcard",
   "warning": ""
 }
-$ ../bitcoin/src/bitcoin-cli -rpcwallet=coldcard importmulti '[{"desc": "wpkh([8038ecd9/84h/0h/0h]xpub6DR4rqx16YnCcfwFqgwvJdKiWrjDRzqxYTY44aoyHwZDSeSB5n2tqt42aYr9qPKhSKUdftPdTjhHrKKD6WGKVbuyhMvGH76VyKKZubg8o4P/0/*)", "internal": false, "range": {"start": 0, "end": 1000}, "timestamp": "now", "keypool": true, "watchonly": true}]'
+$ ../bitcoin/src/bitcoin-cli -rpcwallet=coldcard importmulti '[{"desc": "wpkh([8038ecd9/84h/0h/0h]xpub6DR4rqx16YnCcfwFqgwvJdKiWrjDRzqxYTY44aoyHwZDSeSB5n2tqt42aYr9qPKhSKUdftPdTjhHrKKD6WGKVbuyhMvGH76VyKKZubg8o4P/0/*)", "internal": false, "range": [0, 1000], "timestamp": "now", "keypool": true, "watchonly": true}]'
```
Comment (fanquake) [inline docs/bitcoin-core-usage.md]: These descriptors will need [`checksums`](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums), as they are required for `importmulti` after https://github.com/bitcoin/bitcoin/pull/15368.

Comment (achow101) [reply docs/bitcoin-core-usage.md]: I've added all of the descriptor checksusm.

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin-core/HWI/pull/138/commits/376245ae0f8547d4999dbd0a8efc48ffc043e340

**Pull request merged**